pkg/infrastructure/azure/azure.go

```diff
@@ -135,6 +135,23 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 	p.clientOptions = opts
 	p.computeClientOptions = computeClientOpts
 
+	if err = handleIdentity(ctx, identityInput{
+		installConfig:     installConfig,
+		region:            platform.Region,
+		resourceGroupName: p.ResourceGroupName,
+		subscriptionID:    subscriptionID,
+		tokenCredential:   tokenCredential,
+		infraID:           in.InfraID,
+		clientOpts:        p.clientOptions,
+		tags:              p.Tags,
+	}); err != nil {
+		errMsg := "error creating user-assigned identity: please ensure your user credentials " +
+			"have the User Access Admin Role or if you are not utilizing an Azure Container Registry " +
+			"you can set installconfig.platform.azure.defaultMachinePlatform.identity: None to skip " +
```

Comment:
nit: it should be the `type` right?

```suggestion
			"you can set installconfig.platform.azure.defaultMachinePlatform.identity.type: None to skip " +
```

Also, I guess:
- `installconfig.controlPlane.platform.azure.identity.type`
- `installconfig.compute.platform.azure.identity.type`

will take precedence, so in some cases where controlplane and compute are configured, setting `defaultMachinePlatform` might not be enough?

Reply:
> nit: it should be the type right?

Correct. Thanks for the catch. Fixed in #9739 and included in this PR.

> will take precedence, so in some cases where controlplane and compute are configured, setting defaultMachinePlatform might not be enough?

Yes that's correct. I don't see how to reasonably suggest that in the error message, unless we perhaps make the message more vague to generally say something like "set Identity Type to None for both control plane and compute machine pools in the install config"...

Reply:
Ah, I think the current message sounds good already then :smile:  The `defaultMachinePlatform` is already self-explanatory that it is a "default" that might be overwritten by ctrlplane and compute settings.